epochtalk/epochtalk
# Added basic search capabilities to Admin Panel. Misc Fixes

* Searching by username has been implemented on the Users Management
  View, Users Moderation and Post Moderation Views.

* Bug fix for missing "field" variable in the offCLS method causing
  sort field not to update properly in both moderation views.

* Fix in report resource for post resources pulling from user routes.

* Minor code cleanup in various files.

* Fixed issue with avatar contraint in top bar css.

* URI Encoding key values in pagination directive. This fixes an issue
  when paging through search results with a search string containing
  symbols.

* Updated paging routes to support filtering by search string. Updated
  apidoc markup to reflect these changes.

* Fixed bug with username being sent to the db in users.find not being
  unescaped. This was causing and error when visting users profiles who
  had special characters in their usernames.

Test Procedure:

* Visit admin views User Management, User and Post Moderation and perform a search. Currently you may only search by username. For moderation views search is performed on the offender's username. '%' can be used as a wildcard in your search string.

* The search should be performed on whatever filter you are selected on. So if you have the "Banned" filter selected you will be searching only banned users. If you have the "All" filter selected you will be searching all users. Same applies to the moderation views.

* Clicking the "X" button in the search box will clear the search results and return you to the standard view.

* Upload an avatar that has a greater height than width. The topbar avatar should now be constrained to a square instead of the height of the avatar forcing the topbar to grow.

* Visit the profile of a user with symbols such as '/' in their username, you should now be able to access their profile where as before you would get a 404 error.
